onadata/libs/utils/csv_builder.py

```diff
@@ -551,30 +555,36 @@ def _update_columns_from_data(self, cursor):
             self.ordered_columns[key] = [key] + gps_xpaths
         image_xpaths = [] if not self.include_images \
             else self.dd.get_media_survey_xpaths()
-
-        for record in cursor:
-            # split select multiples
-            if self.split_select_multiples:
-                record = self._split_select_multiples(
-                    record, self.select_multiples,
-                    self.BINARY_SELECT_MULTIPLES, self.VALUE_SELECT_MULTIPLES,
-                    show_choice_labels=self.show_choice_labels)
-            # check for gps and split into components i.e. latitude, longitude,
-            # altitude, precision
-            self._split_gps_fields(record, self.gps_fields)
-            self._tag_edit_string(record)
-            # re index repeats
-            for (key, value) in iteritems(record):
-                self._reindex(
-                    key, value, self.ordered_columns, record, self.dd,
-                    include_images=image_xpaths,
-                    split_select_multiples=self.split_select_multiples,
-                    index_tags=self.index_tags,
-                    show_choice_labels=self.show_choice_labels,
-                    language=self.language)
-
-    def _format_for_dataframe(self, cursor):
-        # TODO: check for and handle empty results
+        # cursor object for xform without instances
+        # is a list of xform headers.
+        if isinstance(cursor, QuerySet):
+            for record in cursor:
+                # split select multiples
+                if self.split_select_multiples:
+                    record = self._split_select_multiples(
+                        record, self.select_multiples,
+                        self.BINARY_SELECT_MULTIPLES,
+                        self.VALUE_SELECT_MULTIPLES,
+                        show_choice_labels=self.show_choice_labels)
+                # check for gps and split into components
+                # i.e. latitude, longitude, altitude, precision
+                self._split_gps_fields(record, self.gps_fields)
+                self._tag_edit_string(record)
+                # re index repeats
+                for (key, value) in iteritems(record):
+                    self._reindex(
+                        key, value, self.ordered_columns, record, self.dd,
+                        include_images=image_xpaths,
+                        split_select_multiples=self.split_select_multiples,
+                        index_tags=self.index_tags,
+                        show_choice_labels=self.show_choice_labels,
+                        language=self.language)
```

Comment:
Are we  handling headers in this function? Could we do without calling this function and hence no need to add the additional check?

Reply:
Sure. This change is breaking existing tests, working to resolve those now

Reply:
Should be good now.